### DISCUSSION TITLE
Any possible way to optimize at build time?

### DISCUSSION THREAD

**Discussion by asertym** [zerodevx/svelte-img#54] [Q&A]
Hey there, 
Thanks for your contribution to the community, the package works great; however I was wondering if we can optimize this at build time. I am currently deploying my project on netlify, and I noticed that build time changed from around 30 seconds to more than 3 minutes due to the images being processed. Is there a way to avoid this issue or speed up the process?

Sorry if this question might not be package related, it's just that I've encountered this issue after I started using it. 
Thanks in advance!

**Comment by molarmanful** [comment]
I've actually been waiting on this for some time now too. The base `vite-imagetools` addressed this in v7 via caching, which would help tremendously in successive runs. `svelte-img` hasn't been upgraded to that yet, but I assume it'll happen at some point.

**Comment by zerodevx** [comment]
Caching is the solution for this. Thing is there are outstanding issues with `vite-imagetools@7` - to be frank it's a bit messy. Basic issues like [this](https://github.com/JonasKruckenberg/imagetools/pull/723) haven't been fixed in months. Users of `v7` are in reality layering transformations atop one another, instead of generating variants from the original source image.

In `v6` and below, the `sharp` instance is merged into the metadata object (that allows us to apply transformations on the original image like make the lqip, retrieve the dominant colour etc.), but not so in `v7` in some instances. Not sure if I want to jump into that rabbit hole.

So until `v7` stablises, we are stuck with `v6`.
